setup.py

```diff
@@ -66,4 +66,11 @@
 
 from extension_helpers import get_extensions  # noqa: E402
 
-setup(ext_modules=get_extensions())
+ext_modules = get_extensions()
+
+# Specify the minimum version for the Numpy C-API
+for ext in ext_modules:
+    if ext.include_dirs and "numpy" in ext.include_dirs[0]:
+        ext.define_macros.append(("NPY_TARGET_VERSION", "NPY_1_21_API_VERSION"))
```

Comment:
Does it crash if the extension code already has its own `NPY_TARGET_VERSION` somehow?

Reply:
Also... so much for getting rid of `setup.py` lol